UnoCheck/Reporting/CheckReportWriter.cs

```diff
@@ -0,0 +1,75 @@
+#nullable enable
+
+using System;
+using System.IO;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace DotNetCheck.Reporting;
+
+internal static class CheckReportWriter
+{
+    private const string ReportFileName = "report.json";
```

Comment:
This should not be written in the appdata, and my cause race conditions on multiple runs. If we need that information, let's pass it as a parameter to uno-check that points to a unique temp file.

Reply:
The benefit to have it in a well known location in app-data is to allow UDEI from all open IDEs to listen to that path to refresh themselves. Also it allows us to include it in an "error reports log package" we are planing to provide.

I would agree it would be great to make sure we don't have concurrent access issues, but as it's only a report I'm fine to keep it simple (as long as we are fault/corruption tolerant on read). Another solution would have been to have all of them in a sub folder and make the file name itself unique, but it would probably require scavenging logic, so IMO we can keep it this way.